pdfplumber/table.py

```diff
@@ -312,7 +312,11 @@ def assign_cell(cell, table):
     if len(current_table["cells"]):
         tables.append(current_table)
 
-    _sorted = sorted(tables, key=lambda t: min(t["corners"]))
+    # Sort the tables identified in order of topmost-and-then-leftmost by sorting via
+    # the Y axis. ``min(t["corners"])`` would give the (X, Y) coordinate of the table's
+    # topleft corner. ``[::-1]`` reverses the tuple so that the sorting happens first on
+    # the Y axis and then on X.
+    _sorted = sorted(tables, key=lambda t: min(t["corners"])[::-1])
```

Comment:
My understanding of this line is that it would do this:

- Find the leftmost-and-then-topmost vertex/corner of each table — `min(t["corners"])`
- Reverse the tuple representation of that specific vertex for each table, so that it is `y,x` instead of `x,y`
- Sort by those reversed tuples

Instead, how about this:

```python
_sorted = sorted(tables, key=lambda t: min(c[::-1] for c in t["corners"]))
```

... which does the reversal before finding the `min` corner? Or perhaps I'm mixed up?

Reply:
Your suggestion is also correct @jsvine but the top left corner of a rectangle will have both the minimum x and y values and hence, both `min(t["corners"])[::-1]` and `min(c[::-1] for c in t["corners"])` should give the same final output provided the rectangle figure in question is upright.
![image](https://user-images.githubusercontent.com/15127115/105309112-e48b4100-5be2-11eb-862c-a1314eeecd81.png)
I didn't consider rotated/skewed rectangles though. Should I?

Reply:
My apologies; I should have described my reasoning more explicitly here. I was thinking less about skewed rectangles, and more about uneven tables, e.g.,:

```
    +---+---+---+
    | A | B | C |
+---+---+---+---+
| E | F | G | H |
+---+---+---+---+
| I | J | K | L |
+---+---+---+---+
```

... where the top-left corner of `A` would be the `topmost-and-then-leftmost` corner, but the top-left corner of `E` would be the `leftmost-and-then-topmost` corner. I admit that this probably a rare case where this would matter ... but it does seem possible.

Reply:
Thanks for the example. The suggestion makes more sense now. Made the change.